lib/gcp/cloudfunctions.js

```diff
@@ -55,13 +55,12 @@ function _updateFunction(options) {
   var data = _.assign({
```

Comment:
failurePolicy is part of EventTrigger. We can honor the existing value by making the updateMask slightly more complicated This _may_ work:

`httpsTrigger,eventTrigger.eventType,eventTrigger.resource,eventTrigger.service`

(presuming you generate the field mask before adding options.trigger to data)

Reply:
I added a commit to use dot notation for updateMask so things like retryPolicy isn't blown away. They do work, I manually tested them.